src/embeddings/embeddings_calculator.cc

```diff
@@ -153,6 +155,20 @@ class EmbeddingsCalculator : public CalculatorBase {
                 SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
                 embeddingsInputMap[embeddingsInputName] = it->second;
             }
+
+            size_t max_context_length = 512;  // default allowed input length. Otherwise, it will be read from model rt_info>config>max_position_embeddings in the model.xml file
+            try {
+                max_context_length = embeddings_session->getModelConfig().at("config").as<ov::AnyMap>().at("max_position_embeddings").as<size_t>();
+            } catch (...) {
+                LOG(INFO) << "Can not read config->max_position_embeddings from model rt_info. Using default value " << max_context_length;
+            }
+
+            size_t input_ids_size = tokenizerOutputMap["input_ids"].get_shape()[1];
```

Comment:
Shall we check if shape looks like expected? (make sure we have second dimension)